src/engine/actions/dml/scan.rs

```
@@ -64,13 +67,25 @@ impl DBEngine {
-    ) -> errors::Result<()> {
+    ) -> errors::Result<usize> {
         if rows.is_empty() {
-            return Ok(());
+            return Ok(0);
         }
 
         let _guard = self.row_storage_lock.lock().await;
         let segment_path = self.row_segment_path(table_name)?;
+
+        // 버퍼 풀의 현재 행 개수를 시작 인덱스로 사용
+        let start_index = self
```

Clippy (1.96.0)

[warning] unnecessary closure used to substitute value for `Option::None`

(warning)